### ISSUE TITLE
IPv6 DHCPv6 DUID-UUID Generation Issue with iPXE on QEMU 8.1.2 and SMBIOS 3.0

### ISSUE THREAD

**Issue by Tharyrok** [ipxe/ipxe#1094]
Hello,

I'm creating this ticket in both projects affected as I'm unsure which side needs to resolve it. I discovered this bug after upgrading Proxmox to version 8.1. I use iPXE to boot in IPv6 and retrieve the configuration from a web server. I have a DHCPv6 and SLAAC server configured.

The versions in use are:
 - QEMU: 8.1.2
 - iPXE commit: f22879ca994aedd2667bb0c73ebce505bf5f8cef
 - SMBIOS version: 3.0

In this configuration, iPXE is unable to generate the necessary DUID-UUID for IPv6. If I revert to the previous QEMU version (using the machine: pc-i440fx-8.0 option in Proxmox), I have no issues. The only difference I notice and understand is the switch to SMBIOS 3.0, which is 64 bits, compared to SMBIOS 2.8, which is 32 bits.

Here's the iPXE trace:

```
iPXE>  ifconf --configurator ipv6
NETDEV net0 opening
NETDEV net0 starting configuration via ipv6
Configuring [ipv6] (net0 66:b5:3e:97:7d:4e)...
NETDEV net0 failed to receive 0x0: Error 0x440e6003 (https://ipxe.org/440e6003)
NETDEV net0 failed to receive 0x0: Error 0x440e6003 (https://ipxe.org/440e6003)
NETDEV net0 failed to receive 0x0: Error 0x440e6003 (https://ipxe.org/440e6003)
DHCPv6 net0 could not create DUID-UUID: No such file or directory (https://ipxe.org/2d0c203b)
NETDEV net0 configuration via ipv6 failed: No such file or directory (https://ipxe.org/2d0c203b)
NETDEV net0 failed to receive 0x0: No such file or directory (https://ipxe.org/2d0c203b)
No such file or directory (https://ipxe.org/2d0c203b)
NETDEV net0 closing
```

What information do you need to debug this situation?

**Comment by Tharyrok** [comment]
Issue on the QEMU side with the reproduction method: https://gitlab.com/qemu-project/qemu/-/issues/2024

**Comment by mcb30** [comment]
@Tharyrok thanks for reporting.  Yes, the issue is that the -8.1 variants provide only the SMBIOS3 entry point structure.

Should be fixed by PR #1113

**Comment by Tharyrok** [comment]
A big thank you @mcb30, the end of December was busy for me, and I didn't take the time to look at this issue. I'm glad to discover that you found the issue and fixed it. In my usage, the correction you provided works perfectly.
